### DISCUSSION TITLE
Mystery and SF participation @ CCC

### DISCUSSION THREAD

**Discussion by mstembera** [official-stockfish/Stockfish#4676] [General]
@AndyGrant 
Hi Andrew,
I am reaching out to you as the person who is in charge of the CCC tournaments at chess.com.  Some time ago a new Mystery engine was introduced as part of an exhibition or bonus series at CCC.  I thought this was a fun way to introduce a new engine but always expected its identity would be revealed after the bonuses.  However, Mystery is now participating in the actual CCC competition still anonymous.  CCC has a policy of letting authors withdraw engines from competition upon request.  Koivisto being a recent example.  I would like to make such a request on behalf of Stockfish.   That is I do not wish to compete against any engine whose author and origin have not been disclosed.  We had issues in the past w/ Houdart cloning SF and passing it off as Houdini as well as Chessbase releasing SF as Fritz.  These were settled in court.  I am NOT accusing Mystery of anything and in fact wish it well.  However while it remains mysterious there is no way to tell.  I should add that I am not a maintainer and only one of many SF developers and contributors.  As such I only speak for myself and don't know to what extent I can make this request.
Thank you.

**Comment by Dboingue** [comment]
maybe using a tool like that on Komodo site to measure distance between 
engine from what i understand to be from some input output behavior 
analysis could help find "phenotypic" relationships with other engines?  
Interesting anecdote this topic.. stood out in my email folder.

Le 11/07/2023 à 20:29, mstembera a écrit :

**Comment by AndyGrant** [comment]
I don't think @mstembera speaks on behalf of the Stockfish team, so while I respect your asking, I will not be making any changes.

Mystery is not a Stockfish fork. You should very well know my position regarding Houdini, Fire, DeusX, Fat Fritz II, and all those around it. You can take my word for it if you like, or not. Perhaps in the future some others will step forward to offer the same commentary I've offered here.

**Comment by mstembera** [comment]
@AndyGrant  I do take your word that Mystery is not a SF fork.  It is really the principle of competing against anonymous engines I have an issue with.  I am simply asking for the courtesy of knowing my opponent.  I do just speak for myself and while I don't know how other contributors feel I suspect that many would feel better if Mystery came forward.  I welcome everyones comments here.

**Reply by AndyGrant** [reply]
You will all know soon enough, don't worry.

**Comment by mstembera** [comment]
@AndyGrant If it's not before we play any more games against Mystery it is not soon enough.   It's is quite biased from a tournament director to give special treatment to one participant.

To other SF contributors:  We suffer a bit from not having a single voice.  We can however use a form of the democratic process to have our collective voices heard. If you would prefer not to play against Mystery until it's author and origin is disclosed please give this a👍.  If not give a 👎.   I promise to abide by the results as I hope does @AndyGrant.

**Reply by AndyGrant** [reply]
I have no interest in a public poll. I've stated above that an announcement is forth coming. And only Vondele's perspective is of interest to CCC decisions.

**Comment by vondele** [comment]
Since I was asked my perspective, I can share. First, I usually don't correspond with CCC on SF participation. There have been interactions on the hardware and SF configuration, but very limited. This is good so, I don't want to be involved in every and any tournament that is out there, and CCC is generally well managed, which is nice as that builds  trust. 

Having said that, these events with "gimmick" engines (e.g. Dragon/Komodo with changed parameters) nor the current mystery engine are something that adds to the standing of the tournament. Imagine that the human chess WC challenger match would have a participant with concealed identity... Additionally, the community has had some bad experiences with engines that concealed their true identity, which is why there is, understandably, some increased sensitivity. However, from the organizer's point of view there can be different objectives of the tournament, and creating some 'hype' around an engine and/or chess can be one. That's not our decision to take.

As such, I see no reason to request withdrawal at the tournament, I'm fine to ignore the hype building aspect, and enjoy the good quality matches otherwise.

**Comment by dav1312** [comment]
> I would like to make such a request on behalf of Stockfish.

> I should add that I am not a maintainer and only one of many SF developers and contributors. As such I only speak for myself and don't know to what extent I can make this request.

This is contradictory. An the one hand you are unilaterally requesting the withdrawal of Stockfish from the competition and on the other hand acknowledging that you are not a maintainer and that you only speak for yourself.

It doesn't make sense that _any_ contributor would have the unilateral power to withdraw the engine they contributed to from competitions. In my opinion you should've asked the community first (and Vondele in particular) before making such request and mentioning Andrew.

**Comment by AlexandreMasta** [comment]
That is what he is trying to do. He is exerting his freedom of speech to do so, that´s the way he thinks and many thinks equal to him (me included).

Do not ask for perfection in "modus operandi" nor expect it from mstembera (he is just human as one of us). That was the way he found to express his concern about competing against anonymous engines and I support him.

> It doesn't make sense that _any_ contributor would have the unilateral power to withdraw the engine they contributed to from competitions. In my opinion you should've asked the community first (and Vondele in particular) before making such request and mentioning Andrew.

**Comment by dav1312** [comment]
To close this discussion: https://www.chess.com/news/view/torch-chess-engine

And from Andrew:
> We had been talking about announcing for a while and the intention was to do it during this current CCC event. The random post here complaining about it had no bearing.

**Comment by mstembera** [comment]
So we just found out that Andrew has been secretly using his position as tournament director to give special privilege to an engine of which he is a coauthor.  Nothing to see here.

**Reply by cj5716** [reply]
In what way has Andrew given privilege to Torch?

**Reply by mstembera** [reply]
No engine has ever had the privilege of competing anonymously.  Also the tournament director participating in the competition is an obvious conflict of interest when not disclosed.

**Reply by dav1312** [reply]
> No engine has ever had the privilege of competing anonymously.

Komodo Dragon played under the exact same pseudonym as Torch before it was released...
https://www.chess.com/computer-chess-championship#event=mystery-engine-debut-and-tuning
https://www.chess.com/news/view/komodo-releases-powerful-new-dragon-chess-engine

**Reply by mstembera** [reply]
I concede I was not aware of that and apologize for making that statement.  Komodo's authors were not the tournament directors however.

**Reply by TheBlackPlague** [reply]
Reiterate though, what difference does it make?

**Reply by mstembera** [reply]
What part of undisclosed conflict of interest should I reiterate?

**Reply by dav1312** [reply]
How can it affect CCC

**Reply by TheBlackPlague** [reply]
The important part: the consequence this undisclosed conflict of interest can impose on CCC and why it would make a difference given Andrew had Ethereal participating in CCC while managing it for years.

**Reply by mstembera** [reply]
That's not for me to speculate.  I prefer to stick to facts.  Now that it's disclosed it's no longer an issue just like Ethereal isn't.  I have nothing further to add.

**Comment by Dboingue** [comment]
chess and secrets!

Le 15/07/2023 à 19:59, mstembera a écrit :

**Comment by ChessOverflow** [comment]
@mstembera, When a chess engine is not open source, How do you recognize if it's a Stockfish fork or not?!

Even the new **Ethereal** version is not open source and it is possible that Stockfish's code has been used in it.

**Reply by cj5716** [reply]
It is open source. https://github.com/AndyGrant/Ethereal

**Reply by ChessOverflow** [reply]
No, **Ethereal 14** which participates in chess tournaments, is not open source.

https://ccrl.chessdom.com/ccrl/4040/cgi/engine_details.cgi?eng=Ethereal%2014.00%2064-bit%204CPU

The open source version of  Ethereal is weak and it has a low rating.

**Reply by cj5716** [reply]
Nope, the version that participates in chess tournaments is the latest dev version of Ethereal, Ethereal 14.17. Ethereal 14 was closed source, but Ethereal is no longer closed source. The only thing is that CCRL mostly tests official releases which is why you see Ethereal 14.00 being listed there.
Ethereal 14.17 is stronger than Ethereal 14.00 and not weaker.

**Reply by cj5716** [reply]
Ethereal's last release was commercial, but it has since turned open source. The one you see on GitHub is the one competing in tournaments such as CCC and TCEC

**Reply by AndyGrant** [reply]
Neither of you have accurately portrayed Ethereal here. 

Ethereal is open-source, and it has never been anything but open-source. However, if you want the NNUE weights, that is a commercial product. This is why Ethereal is Blue on CCRL, despite being open-source. **Its both open-source and commercial at the same time.**

Ethereal's Networks are trained with a CPU trainer I wrote found here ( https://github.com/AndyGrant/NNTrainer ). Its fairly good, I get 350k NPS on L1=1536 architectures, but this is nothing compared to what I could get if I could utilize my GPU. 

All data is generated via cutechess, stripped down to a minimal format via a python script, and then converted into 32-byte-per-position data directly in a SAN parser inside of Ethereal. That data then gets shuffled via ( https://gist.github.com/AndyGrant/b3a70ea5f5ff4285aa60076381ee30c4 ), and fed into the trainer.

Up until this most recent version, all of my data is strictly Ethereal vs Ethereal. However my most recent Network included Stockfish _as an adversary_, meaning I play against Stockfish, but I don't use the data produced by Stockfish. You can see some of my recent changes in [this commit](https://github.com/AndyGrant/Ethereal/commit/29003c79a51742a4b0df06bdd4e8807fcf0b1e42) to make this possible.

Please end this conversation. I'm tired of having people share this thread with me.

**Reply by ChessOverflow** [reply]
Can you prove that commercial and open-source version of Ethereal are the same?

Is the `bench` the same in both of them if NNUE is disabled?

Is it possible to change or disable NNUE in the commercial version?

**Reply by AndyGrant** [reply]
Buy a copy and reverse engineer the binary. Unsubscribed from this noise. Have real work to do.

**Reply by TheBlackPlague** [reply]
Retardness knows no bounds. Blocking @ChessOverflow and unsubscribing from this nonsense.
